indico/modules/events/papers/operations.py

```diff
@@ -125,3 +126,39 @@ def judge_paper(contribution, contrib_data, judgment, judge, send_notifications=
     contribution.event_new.log(EventLogRealm.management, EventLogKind.change, 'Papers',
                                'Paper "{}" was judged'.format(orig_string(contribution.verbose_title)), judge,
                                data=log_data)
+
+
+def _store_paper_template_file(template, file):
+    content_type = mimetypes.guess_type(file.filename)[0] or file.mimetype or 'application/octet-stream'
+    filename = secure_filename(file.filename, 'template')
+    # reset fields in case an existing file is replaced so we can save() again
+    template.storage_backend = None
+    template.storage_file_id = None
+    template.size = None
+    template.content_type = content_type
+    template.filename = filename
+    template.save(file.file)
+
+
+def create_paper_template(event, data):
+    file = data.pop('template')
+    template = PaperTemplate(event_new=event)
+    template.populate_from_dict(data)
+    _store_paper_template_file(template, file)
+    db.session.flush()
+    logger.info('Paper template %r uploaded by %r', template, session.user)
```

Comment:
Why differentiate the user's feedback from Indico when there are many or not many of the kind they are adding/removing/updating?